src/ttvchat/AuthenticationService.ts

```diff
@@ -0,0 +1,121 @@
+      if (!accessToken) {
+        const state = v4();
+        this.createServer(state);
+        env.openExternal(Uri.parse(`https://id.twitch.tv/oauth2/authorize?client_id=${TwitchKeys.clientId}` +
+          `&redirect_uri=http://localhost:5544` +
+          `&response_type=token&scope=${TwitchKeys.scope}` +
+          `&force_verify=true` +
+          `&state=${state}`));
+      }
+      else {
+        const validResult = await API.validateToken(accessToken);
+        if (validResult.valid) {
+          this._onAuthStatusChanged.fire(true);
+        }
+      }
+    }
+  }
+
+  public async signOutHandler() {
+    if (keytar) {
+      const token = await keytar.getPassword(KeytarKeys.service, KeytarKeys.account);
+      if (token) {
+        const revoked = await API.revokeToken(token);
+        if (revoked) {
+          window.showInformationMessage('Twitch token revoked successfully');
+        }
+      }
+      keytar.deletePassword(KeytarKeys.service, KeytarKeys.account);
+      keytar.deletePassword(KeytarKeys.service, KeytarKeys.userLogin);
+    }
+    this._onAuthStatusChanged.fire(false);
+  }
+
+  private createServer(state: string) {
+    const filePath = path.join(extensions.getExtension(extensionId)!.extensionPath, 'out', 'ttvchat', 'login', 'index.htm');
+    this.log(LogLevel.Debug, `Starting login server using filePath: ${filePath}.`);
+    const file = readFileSync(filePath);
+    if (file) {
+      const server = http.createServer(async (req, res) => {
+        const mReq = url.parse(req.url!, true);
+        const mReqPath = mReq.pathname;
+
+        if (mReqPath === '/') {
+          res.writeHead(200, { 'Content-Type': 'text/html' });
+          res.end(file);
+        }
+        else if (mReqPath === '/oauth') {
+          const q: any = mReq.query;
+
+          if (q.state !== state) {
+            window.showErrorMessage('Error while logging in. State mismatch error.');
+            await API.revokeToken(q.access_token);
+            this._onAuthStatusChanged.fire(false);
+            res.writeHead(500, 'Error while logging in. State mismatch error.');
+            res.end();
+            return;
+          }
+
+          res.writeHead(200);
+          res.end(file);
+
+          const validationResult = await API.validateToken(q.access_token);
+          if (keytar && validationResult.valid) {
+            keytar.setPassword(KeytarKeys.service, KeytarKeys.account, q.access_token);
+            keytar.setPassword(KeytarKeys.service, KeytarKeys.userLogin, validationResult.login);
+            this._onAuthStatusChanged.fire(true);
+          }
+        }
+        else if (mReqPath === '/complete') {
+          res.writeHead(200);
+          res.end(file);
+          setTimeout(() => server.close(), 3000);
+        }
+        else if (mReqPath === '/favicon.ico') {
+          res.writeHead(204);
+          res.end();
+        }
+      });
+
+      server.listen('5544', (err: any) => {
```

Comment:
Is this a required port or can it be dynamic or? Just wondering if it should be hard-coded or moved to an env variable or something.